imap-proto/src/types.rs

```diff
@@ -293,6 +297,7 @@ pub enum Attribute {
     Rfc822Size,
     Rfc822Text,
     Uid,
+    XGmLabels,
```

Comment:
Should have a comment linking to https://developers.google.com/gmail/imap/imap-extensions#access_to_gmail_labels_x-gm-labels.

Reply:
I propose naming the enum `GmailLabels` following https://github.com/djc/tokio-imap/pull/138#discussion_r775630683.